rustls-post-quantum/src/lib.rs

```diff
@@ -43,214 +43,59 @@
-        })
-    }
-
-    fn ffdhe_group(&self) -> Option<FfdheGroup<'static>> {
-        None
-    }
-
-    fn name(&self) -> NamedGroup {
-        NamedGroup::X25519MLKEM768
-    }
-
-    fn usable_for_version(&self, version: ProtocolVersion) -> bool {
-        version == ProtocolVersion::TLSv1_3
-    }
-}
-
-struct Active {
-    x25519: Box<dyn ActiveKeyExchange>,
-    decap_key: Box<kem::DecapsulationKey<kem::AlgorithmId>>,
-    combined_pub_key: Vec<u8>,
-}
-
-impl ActiveKeyExchange for Active {
-    fn complete(self: Box<Self>, peer_pub_key: &[u8]) -> Result<SharedSecret, Error> {
-        let Some(ciphertext) = ReceivedCiphertext::new(peer_pub_key) else {
-            return Err(INVALID_KEY_SHARE);
-        };
-
-        let combined = CombinedSecret::combine(
-            self.x25519
-                .complete(ciphertext.x25519)?,
-            self.decap_key
-                .decapsulate(ciphertext.ml_kem.into())
-                .map_err(|_| INVALID_KEY_SHARE)?,
-        );
-        Ok(SharedSecret::from(&combined.0[..]))
-    }
-
-    /// Allow the X25519 computation to be offered and selected separately.
-    fn hybrid_component(&self) -> Option<(NamedGroup, &[u8])> {
-        Some((self.x25519.group(), self.x25519.pub_key()))
-    }
-
-    fn complete_hybrid_component(
-        self: Box<Self>,
-        peer_pub_key: &[u8],
-    ) -> Result<SharedSecret, Error> {
-        self.x25519.complete(peer_pub_key)
-    }
-
-    fn pub_key(&self) -> &[u8] {
-        &self.combined_pub_key
-    }
-
-    fn ffdhe_group(&self) -> Option<FfdheGroup<'static>> {
-        None
-    }
-
-    fn group(&self) -> NamedGroup {
-        NamedGroup::X25519MLKEM768
-    }
-}
-
-struct ReceivedShare<'a> {
-    ml_kem: &'a [u8],
-    x25519: &'a [u8],
-}
-
-impl<'a> ReceivedShare<'a> {
-    fn new(buf: &'a [u8]) -> Option<ReceivedShare<'a>> {
-        if buf.len() != COMBINED_PUBKEY_LEN {
-            return None;
-        }
-
-        let (ml_kem, x25519) = buf.split_at(MLKEM768_ENCAP_LEN);
-        Some(ReceivedShare { ml_kem, x25519 })
-    }
-}
-
-struct ReceivedCiphertext<'a> {
-    ml_kem: &'a [u8],
-    x25519: &'a [u8],
-}
-
-impl<'a> ReceivedCiphertext<'a> {
-    fn new(buf: &'a [u8]) -> Option<ReceivedCiphertext<'a>> {
-        if buf.len() != COMBINED_CIPHERTEXT_LEN {
-            return None;
-        }
-
-        let (ml_kem, x25519) = buf.split_at(MLKEM768_CIPHERTEXT_LEN);
-        Some(ReceivedCiphertext { ml_kem, x25519 })
-    }
-}
-
-struct CombinedSecret([u8; COMBINED_SHARED_SECRET_LEN]);
-
-impl CombinedSecret {
-    fn combine(x25519: SharedSecret, ml_kem: kem::SharedSecret) -> Self {
-        let mut out = CombinedSecret([0u8; COMBINED_SHARED_SECRET_LEN]);
-        out.0[..MLKEM768_SECRET_LEN].copy_from_slice(ml_kem.as_ref());
-        out.0[MLKEM768_SECRET_LEN..].copy_from_slice(x25519.secret_bytes());
-        out
-    }
-}
-
-struct CombinedShare(Vec<u8>);
+pub static X25519MLKEM768: &dyn SupportedKxGroup = &hybrid::Hybrid {
```

Comment:
Consider changing the order of commits to avoid the intra-PR changes? Can always add Deirdre as a co-author on your commits.

Reply:
Put these two commits into one.